### PR TITLE
[kaidjohnson] Fix up dateparser support for 12 hour datetime format.

### PR DESCRIPTION
**Pull request by kaidjohnson**
Using a date format that includes a 12-hour time format (eg. dd MMMM yyyy hh:mma) can result in a permanent invalid parsing of the value.

I've created a plunk to demonstrate the issue (try manually deleting the 'M' from 'PM' then putting the 'M' back. It becomes invalid without the M but never is valid again when the M is restored.):
http://plnkr.co/edit/CvJgnoCE08XGlDbVQka8?p=preview

Here's the same plunkr, but with the fix found in this PR:
http://plnkr.co/edit/uTKEjRFcY3CsWDHyZ0bZ?p=preview

### CODE REVIEW

**Comment by icfantv** [comment]
@kaidjohnson please add the relevant functional tests for this and update/rebase your PR.  thanks.

File: src/dateparser/dateparser.js [hunk 1]
```diff
@@ -53,12 +53,16 @@ angular.module('ui.bootstrap.dateparser', [])
...
+      regex: '0[0-9]{1}|1[0-2]{1}',
+      apply: function(value) { this.hours = +value; }
+    },
     'H': {
       regex: '1?[0-9]|2[0-3]',
       apply: function(value) { this.hours = +value; }
     },
     'mm': {
-      regex: '[0-5][0-9]',
+      regex: '[0-5]{1}[0-9]{1}',
```
Comment (wesleycho) [inline src/dateparser/dateparser.js]: Revert this change - this is no different than the previous version, as tested [here](http://regexpal.com/).

File: src/dateparser/dateparser.js [hunk 2]
```diff
@@ -53,12 +53,16 @@ angular.module('ui.bootstrap.dateparser', [])
       regex: '(?:0|1)[0-9]|2[0-3]',
       apply: function(value) { this.hours = +value; }
     },
+    'hh': {
+      regex: '0[0-9]{1}|1[0-2]{1}',
```
Comment (wesleycho) [inline src/dateparser/dateparser.js]: Remove `{1}` in both instances in this line, they are unnecessary.

**Comment by kaidjohnson** [comment]
Thanks @icfantv and @wesleycho -- I've added tests, removed redundant {1}, and corrected AM/PM parsing issue discovered during tests. Also updated plunkr

**Comment by wesleycho** [comment]
LGTM - good work!
